### PR TITLE
Fixed hunter pets not being revived inside battlegrounds.

### PR DESCRIPTION
**Pull request by ChrisCGalbraith**
Fixed hunter pets not having max health when player is resurrected in battlegrounds.

Signed-off-by: ChrisCGalbraith <chriscgalbraith@gmai.com>

## Changes Proposed:
-  New summon of the players pet occurs when revived by the battleground spirit healer

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/1193

## Tests Performed:

- Performed tests on hunters and warlocks with pets that are dead at the time of resurrection and pets that have a portion of their health missing.
- Built on Windows 10, no errors.

## How to Test the Changes:

1. .debug bg on
2. Enter a battleground and wait for it to begin
3. .die your pet and then yourself
4.  Observe that pet is revived at full health when the player is.
5.  Repeat the above steps but instead of killing your pet, .damage him

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Battlegrounds/Battleground.cpp [hunk 1]
```diff
@@ -365,8 +365,14 @@ inline void Battleground::_ProcessResurrect(uint32 diff)
             Player* player = ObjectAccessor::FindPlayer(guid);
             if (!player)
                 continue;
+            Pet*  pet = player->GetPet();
```
Comment (Kitzunu) [inline src/server/game/Battlegrounds/Battleground.cpp:368]: remove double space and add a newline between this and the if statement above.

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Comment by Nyeriah** [comment]
This is wrong. Players use spell 6962 to summon pets, and it's scripted here: https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/scripts/Spells/spell_generic.cpp#L2234

Right solution would be to find why those lines are failing: https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/scripts/Spells/spell_generic.cpp#L2251-L2252

But it seems to me Trinity has it working and we just need a port: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Spells/spell_generic.cpp#L2885-L2914

**Comment by Nyeriah** [comment]
> > This is wrong. Players use spell 6962 to summon pets, and it's scripted here: https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/scripts/Spells/spell_generic.cpp#L2234
> > Right solution would be to find why those lines are failing: https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/scripts/Spells/spell_generic.cpp#L2251-L2252
> > But it seems to me Trinity has it working and we just need a port: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/scripts/Spells/spell_generic.cpp#L2885-L2914
> 
> Cheers for pointing this out - I noticed that the hunter receives a spell failed as the target is dead error when casting 6962. I'll investigate this some more. While this pull request does solve the issue, shall I close this PR in favour of developing a better solution in line with your comment?

You don't need to close it, you can just edit the files and push to this branch, it'll automatically update the PR.

The source I linked there from the TrinityCore repository is a good hint. They're doing a lot of things to pets we currently don't (like reseting health etc.)

**Comment by bgcfb** [comment]
The pr not fix Wintergrasp
just add code to game\Battlefield\Battlefield.cpp L743

**Comment by Nyeriah** [comment]
@bgcfb This is a work-in-progress PR. Indeed, it won't affect Wintergrasp. Thanks for checking it out, but the author is working on updates that will eventually address that issue. Please test it again once this PR is updated.

**Comment by slicplaya** [comment]
This issue has not been fixed.

**Pull request closed without merging**